zlib-rs/src/deflate/window.rs

```diff
@@ -91,52 +91,9 @@ impl<'a> Window<'a> {
     // memory at the end of the window so that chunked operations can use the whole buffer. If they
     // go slightly over `self.capacity` that's okay, we account for that here by making sure the
     // memory there is initialized!
-    pub fn initialize_out_of_bounds(&mut self) {
-        const WIN_INIT: usize = crate::deflate::STD_MAX_MATCH;
-
-        // If the WIN_INIT bytes after the end of the current data have never been
-        // written, then zero those bytes in order to avoid memory check reports of
-        // the use of uninitialized (or uninitialised as Julian writes) bytes by
-        // the longest match routines.  Update the high water mark for the next
-        // time through here.  WIN_INIT is set to STD_MAX_MATCH since the longest match
-        // routines allow scanning to strstart + STD_MAX_MATCH, ignoring lookahead.
-        if self.high_water < self.capacity() {
-            let curr = self.filled().len();
-
-            if self.high_water < curr {
-                // Previous high water mark below current data -- zero WIN_INIT
-                // bytes or up to end of window, whichever is less.
-                let init = Ord::min(self.capacity() - curr, WIN_INIT);
-
-                self.buf.as_mut_slice()[curr..][..init].fill(0);
-
-                self.high_water = curr + init;
-
-                self.filled += init;
-            } else if self.high_water < curr + WIN_INIT {
-                // High water mark at or above current data, but below current data
-                // plus WIN_INIT -- zero out to current data plus WIN_INIT, or up
-                // to end of window, whichever is less.
-                let init = Ord::min(
-                    curr + WIN_INIT - self.high_water,
-                    self.capacity() - self.high_water,
-                );
-
-                self.buf.as_mut_slice()[self.high_water..][..init].fill(0);
-
-                self.high_water += init;
-                self.filled += init;
-            }
-        }
-    }
+    pub fn initialize_out_of_bounds(&mut self) {}
 
-    pub fn initialize_at_least(&mut self, at_least: usize) {
-        let end = at_least.clamp(self.high_water, self.buf.len());
-        self.buf.as_mut_slice()[self.high_water..end].fill(0);
-
-        self.high_water = end;
-        self.filled = end;
-    }
+    pub fn initialize_at_least(&mut self, at_least: usize) {}
```

Comment:
This commit in isolation would cause the `.filled()` method to return a shorter output as you removed `self.filled = end;`. Because of the next commit the PR as a whole doesn't have this problem though.